### PR TITLE
DEV: Switch to web-crypto-based TOTP library, remove JQuery

### PR DESCRIPTION
**Pull request by davidtaylorhq**
Switching to a much smaller (17kb -> 3kb) TOTP implementation also means we can afford to load the code in the main theme JS bundle. This avoids the need to defer loading the library via `loadScript`, and makes this theme component work properly on CSP-equipped sites.

### CODE REVIEW

File: javascripts/discourse/initializers/discourse-otp.js.es6 [hunk 1]
```diff
@@ -1,34 +1,34 @@
 import { withPluginApi } from "discourse/lib/plugin-api";
-import loadScript from "discourse/lib/load-script";
+import { later } from "@ember/runloop";
+import TinyTOTP from "../vendor/tiny-totp";
```
Comment (udan11) [inline javascripts/discourse/initializers/discourse-otp.js.es6:3]: This is nice. :+1:

File: javascripts/vendor/tiny-totp.js [hunk 2]
```diff
@@ -0,0 +1,158 @@
+/*
```
Comment (udan11) [inline javascripts/vendor/tiny-totp.js:1]: Should this be added to `package.json` and add it to `lib/tasks/javascript.rake` too?

**Review by udan11 — APPROVED** [review]
Looks good to me. I left just a couple of minor comments. :+1:

**Comment by SamSaffron** [comment]
yay!

**Pull request merged**